lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,248 @@
+import { exec } from 'child_process';
+import { createHash } from 'crypto';
+import { createReadStream, existsSync, mkdirSync, statSync } from 'fs';
+import { performance } from 'perf_hooks';
+import readline from 'readline';
+import { promisify } from 'util';
+import { GetReleasesConfig, ReleaseResult } from '..';
+import { logger } from '../../logger';
+import { Datasource } from '../datasource';
+import { DebLanguageConfig, PackageDescription } from './types';
+
+/**
+ * @brief This datasource allows to download
+ */
+export class DebDatasource extends Datasource {
+  static readonly id = 'deb';
+
+  constructor() {
+    super(DebDatasource.id);
+  }
+
+  /**
+   * Users are able to specify custom Debian repositories as long as they follow
+   * the Debian package repository format as specified here
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   */
+  override readonly customRegistrySupport = true;
+
+  /**
+   * deb uri distribution [component1] [component2] [...]
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   */
+  override readonly defaultRegistryUrls = [
+    'deb https://ftp.debian.org/debian stable main contrib non-free',
+  ];
+
+  override readonly defaultConfig: DebLanguageConfig = {
+    deb: {
+      binaryArch: 'amd64',
+      downloadDirectory: '/tmp/renovate-deb/packages-download',
+      extractionDirectory: '/tmp/renovate-deb/packages',
```

Comment:
nope, use `ensureCacheDir` from `util/fs`, checkout `terraform manager for sample usage

https://github.com/renovatebot/renovate/blob/4f65b57225a42f9deb010f8dbfeb06eb94c0dd69/lib/manager/terraform/lockfile/hash.ts#L93-L102

Reply:
Done